client/src/core/SurveyPopup.js

```diff
@@ -0,0 +1,130 @@
+import text from "./SurveyPopup.yaml";
+
+import { withRouter } from 'react-router';
+
+import { FixedPopover, create_text_maker_component } from '../components/index.js';
+
+const {
+  TM,
+  text_maker,
+} = create_text_maker_component(text);
+
+
+const get_path_root = (path) => _.chain(path)
+  .replace(/^\//, '')
+  .split('/')
+  .first()
+  .value();
+
+const seconds_in_a_half_year = 60*60*24*(365/2);
+const should_reset_local_storage = () => localStorage.getItem(`infobase_survey_popup_deactivated`) &&
+  Date.now() - localStorage.getItem(`infobase_survey_popup_deactivated_since`) > seconds_in_a_half_year;
+
+const get_state_defaults = () => {
+  const default_active = true;
+  const default_chance = 0;
+
+  const local_storage_deactivated = localStorage.getItem(`infobase_survey_popup_deactivated`);
+  const local_storage_chance = localStorage.getItem(`infobase_survey_popup_chance`);
+
+  // localStorage is all strings, note that we cast the values read from it to a boolean and a number below
+  return {
+    active: !_.isNull(local_storage_deactivated) ? !local_storage_deactivated : default_active,
+    chance: !_.isNull(local_storage_chance) ? +local_storage_chance : default_chance,
+  };
+};
+
+
+const chance_increment = 0.15;
```

Comment:
So I was thinking about this last night, and I wonder if we are overestimating the amount that people will click around the infobase. I think we should aim to show the popup to most users (since the vast majority will not fill it out). How many different routes do we think most users visit? Home + 1, maybe 2 infographics? If so then most users won't see this.

Reply:
Hmm, it's a tricky balance to strike, between showing it enough but not spamming it at people who haven't seen enough pages to have any comments yet (in which case I assume they dismiss it right away, even if they would then later have potential comments).

I'll make it more likely though.